e3db/tests/test_search_integration.py

```diff
@@ -182,25 +181,163 @@ def test_v2_search_fuzzy(self):
+        end = datetime.now().astimezone()
+        q = e3db.types.Search(include_data=True).match(record_types=[self.record_type]) \
+            .range(end=end)
         results = self.client1.search(q)
         assert(len(results)==2)
 
-    def test_v2_valid_range(self):
-        q = e3db.types.Search(include_data=True).match(record_types=[self.record_type]).range(zone="PST", start=datetime.now()+timedelta(hours=-1), end=datetime.now()+timedelta(hours=1))
+    def test_v2_valid_range_offset(self):
+        start = datetime.now().astimezone()+timedelta(hours=-1)
+        end = datetime.now().astimezone()+timedelta(hours=1)
+        q = e3db.types.Search(include_data=True).match(record_types=[self.record_type]) \
+            .range(start=start, end=end)
         results = self.client1.search(q)
         assert(len(results)==2)
 
+    def test_v2_invalid_range_offset(self):
+        start = datetime.now()+timedelta(hours=-1)
+        end = datetime.now()+timedelta(hours=1)
+        q = e3db.types.Search(include_data=True).match(record_types=[self.record_type]) \
+            .range(start=start, end=end)
+        results = self.client1.search(q)
+        assert(len(results)==0)
+
+    def test_v2_valid_range_unix_search(self):
+        unix_now = int(time.time())
+        start = unix_now-1000
+        end = unix_now+1000
+        q = e3db.types.Search(include_data=True).match() \
+            .range(start=start, end=end)
+        results = self.client1.search(q)
+        assert(len(results)==2)
+
+    def test_v2_valid_range_zoneoffset_search(self):
+        start = datetime.now()+timedelta(hours=-2)
+        end = datetime.now()+timedelta(hours=2)
+        q = e3db.types.Search(include_data=True).match(record_types=[self.record_type]) \
+            .range(start=start, end=end, zone_offset="-07:00")
+        results = self.client1.search(q)
+        assert(len(results)==2)
+
+        q = e3db.types.Search(include_data=True).match() \
+            .range(start=start, end=end, zone_offset=-7)
+        results = self.client1.search(q)
+        assert(len(results)==2)
+
+    def test_v2_range_timezone_conversion_use_local_timezone(self):
+        local_datetime = datetime.now().astimezone()
+        start = local_datetime
+        end = local_datetime
+        r = types.Range(start=start, end=end)
+        assert(r.start == local_datetime)
+        assert(r.end == local_datetime)
+
+    def test_v2_range_timezone_conversion_default_start_UTC(self):
+        naive_now  = datetime.now() 
+        start = naive_now
+        r = types.Range(start=start)
+        assert(r.start != naive_now.astimezone(timezone(timedelta(0))))
+        assert(r.start_formatted()[-6:] == "+00:00")
+        assert(r.end is None)
+
+    def test_v2_range_timezone_conversion_default_end_UTC(self):
+        naive_now = datetime.now() 
+        end = naive_now
+        r = types.Range(end=end)
+        assert(r.end != naive_now.astimezone(timezone(timedelta(0))))
```

Comment:
Will this pass in a build environment with a clock set to UTC?

Reply:
It wouldn't have in that scenario, altered to make sure it will pass.